ext/js/pages/settings/text-transformations-controller.js

```diff
@@ -0,0 +1,171 @@
+/*
+ * Copyright (C) 2024  Yomitan Authors
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+
+import {querySelectorNotNull} from '../../dom/query-selector.js';
+
+export class TextTransformationsController {
+    /**
+     * @param {import('./settings-controller.js').SettingsController} settingsController
+     */
+    constructor(settingsController) {
+        /** @type {import('./settings-controller.js').SettingsController} */
+        this._settingsController = settingsController;
+        /** @type {HTMLSelectElement} */
+        this._container = querySelectorNotNull(document, '#text-transformations');
+        /** @type {import('language').TextTransformation[]}*/
+        this._transformations = [];
+        /** @type {string} */
```

Comment:
Make this a nullable so there's never a chance of accidentally setting the empty string language settings or something like that.